addons/test_mail/tests/test_mail_activity.py

```diff
@@ -225,6 +226,39 @@ def test_activity_summary_sync(self):
             # activity summary remains unchanged from change of activity type as call activity doesn't have default summary
             self.assertEqual(ActivityForm.summary, email_activity_type.summary)
 
+    @mute_logger('odoo.sql_db')
+    @users('employee')
+    def test_activity_values(self):
```

Comment:
As mentioned, it would be ideal to test the "newId" use case if possible.

Reply:
Added a test and indeed, on a newID it generates an activity with res_id = 0 .